ambry-network/src/main/java/com.github.ambry.network/Transmission.java

```diff
@@ -132,8 +132,7 @@ public SocketAddress getRemoteSocketAddress() {
   /**
    * Close the connection for the socket channel
    */
-  public abstract void close()
-      throws IOException;
```

Comment:
Also, like i said "how can we justify that the abstract method needs to throw IOException, when none of the inherited methods do" is not a good argument. The abstract method is the one that defines the contract IMO - whether the implementation throws or not is an implementation detail. What you need to think about is whether throwing makes sense for the what the method does and I think it does in this case.